### DISCUSSION TITLE
Explore an effective way to track "necessary" content difference for l10n contributors

### DISCUSSION THREAD

**Discussion by awxiaoxian2020** [mdn/mdn-community#160] [Localization] [answered]
**Problem**:
Now, we have a lot to do for `translated content`. For example, convert html to md, some macros and etc. But when we have done these chores in the future, for l18n contributors, the main work may be syncing the `content`. Thus, an effective way to tell the contributors **necessary** difference , as @SphinxKnight said: 

> Some of which might be simple typos from English (with a priori no impact on localized pages), some might be 'technical' updates (batch search / replace) which could be applied similarly in mdn/translated-content and finally, the most interesting ones (also the most complex/time-consuming) are 'real' content updates.

Yes, we have a dashboard to see the time and some other differences, but:
- For example, when I visit http://localhost:3000/zh-CN/_translations/missing , it doesn't list any missing translation **pages** for us. we know there are many pages only exist in English versions, we should track them.
- we don't expect that, when the `update time` updated in `translated content`, and then we check `translated content` to decide whether to update it. And sometimes `translated content` earlier than `content`, **l10n contributors may be for other typos not the syncs**. According to the time is not a good idea.

**A possible solution**:
we add the options for "Do these changes need to sync with l10n versions?" in PR template in `content` PRs. 
>If possible, it could be checked by CI(when the option didn't be noted by contributors or reviewers, we will not pass the ci). If not possible, we believe that some contributors and reviewers are careful. We can track some at least.

We assume there will be an issue in every l18n project. When option is `yes`, bot will record it in some way. And weekly comment a task list of the changes that wasn't synced under the issue. when synced, the reviewer edits the task list for completed.

**Comment by fwqaaq** [comment]
I'm on your side. Outdated translations are a real hassles.

**Comment by fwqaaq** [comment]
* This is that I thought, when `mdn content` changes, can you raise a issue in `content tranlation`?
* Maybe we can take full advantages of github acitons.

**Reply by yin1999** [reply]
> This is that I thought, when mdn content changes, can you raise a issue in content tranlation?

This may be a bit annoying for there are a lot of typo fixes in `mdn/content`

**Reply by SphinxKnight** [reply]
In terms of "volume", what would happen in the following case:
- an update is made in English on document A thus, triggering the creation of a new PR
- <because any reason, the update is not applied quickly on the localized content>
- another update is made in English on the same document A

Should this trigger the creation of two PRs? Should the GitHub Actions (or any other mechanism) be "smart enough" to close the previous one?
How is the localization team supposed to handle this? Closing the first one? Dealing with both?

@sideshowbarker is keeping tracks of indicators for `mdn/content` and may help giving an idea of how many PRs are merged per day on `mdn/content` (if I understand correctly, such an automatic system would create a similar quantity of PRs on mdn/translated-content).

**Reply by fwqaaq** [reply]
> > This is that I thought, when mdn content changes, can you raise a issue in content tranlation?
> 
> This may be a bit annoying for there are a lot of typo fixes in `mdn/content`

Maybe we need to write a `github acition` ourselves to campare their differences

**Reply by SphinxKnight** [reply]
OK so here are the numbers that @sideshowbarker reported earlier on Matrix/Element for mdn/content
>I notice we’ve all had a really busy month at mdn/content — looking at output from my reporting tool, I see that during the last 4 weeks we had:
>  - 880 commits in total (vs monthly average of 626)
>  - 855 PRs merged (vs monthly average of 615)
>  - 226 people committed (vs monthly average of 183)
>  - 140 of those people were first-time committers (vs monthly average of 126)
>  - 240 issues closed (vs monthly average of 212)

**Reply by SphinxKnight** [reply]
I'll only speak for @mdn/yari-content-fr here but as of now, we can't afford *just triaging* more than 600 ~PRs~issues* per month.

**Reply by awxiaoxian2020** [reply]
@SphinxKnight my solution seems don't need a new PR or issue, only comments and labels or others.
(sorry, I give a wrong information. I only say there should be an issue to track all sync PRs by comments. The comment need not every day, can be weekly)

**Reply by SphinxKnight** [reply]
@awxiaoxian2020 yes, sorry, I replied in this thread for the "raise an issue" option.

Yes, having a weekly/monthly report of updated docs (given the last contributions on mdn/content) is a nice alternative.

**Comment by yin1999** [comment]
> we add the options for "Do these changes need to sync with l10n versions?" in PR template in `content` PRs.

We should consider whether all the l10n teams can deal with those content changes as quickly as possible.

**Reply by awxiaoxian2020** [reply]
> > we add the options for "Do these changes need to sync with l10n versions?" in PR template in `content` PRs.
> 
> We should consider whether all the l10n teams can deal with those content changes as quickly as possible.

If `yes`, just a comment under the PR and a comment under the sticky issue in l18n project. If they don't want to change it right now, just as it is.

**Comment by awxiaoxian2020** [comment]


**Reply by fwqaaq** [reply]
> @Jack-Zhang-1314 @SphinxKnight @yin1999 I update the solution. PTAL. :)

* the sticky issue refers to the following ?
![](https://user-images.githubusercontent.com/82551626/181175649-7afcf6e3-ccab-41da-953f-138a30a94a57.png)

**Reply by awxiaoxian2020** [reply]
> > @Jack-Zhang-1314 @SphinxKnight @yin1999 I update the solution. PTAL. :)
> 
> * the sticky issue refers to the following ?
>   ![](https://user-images.githubusercontent.com/82551626/181175649-7afcf6e3-ccab-41da-953f-138a30a94a57.png)

It can be pinned, I only want to say it's always open not closed.

**Reply by fwqaaq** [reply]
I see, you mean that content in the issue is a link to the `content` pr ?

**Reply by awxiaoxian2020** [reply]
> I see, you mean that content in the issue is a link to the `content` pr ?

Yes.

**Reply by fwqaaq** [reply]
maybe this is a good👍🏻 idea.

**Comment by awxiaoxian2020** [comment]


**Reply by fwqaaq** [reply]
It is marked with the last modification date at the bottom of the page.But these are unmaintainable practices and it is hard to find the difference.

**Comment by awxiaoxian2020** [comment]
Hello, everyone! I have a new solution which is simple. When we have a `content` Pull Request about the `.md` file, we add a label like `need sync`. If the PR is only typos or others which should not be synced, the reviewer can remove it. Then the l10n contributors can use the label filter to contribute!

**Reply by fwqaaq** [reply]
Maybe you can look at this https://github.com/mdn/translated-content/issues/8918

**Comment by awxiaoxian2020** [comment]
https://github.com/mdn/translated-content/blob/3d99ffdd07d1c37d16badfc35f35379fb87decf2/docs/README.md?plain=1#L23 seems a good idea if we can display the `hash` difference in dashboard.

Maybe related to https://github.com/orgs/mdn/discussions/129.

We can get `l10n.sourceCommit` from translated files, and compare it with English content commit. Is it difficult to implement?

**Reply by yin1999** [reply, marked as answer]
Initial ideas have been implemented here: mdn/yari#8338
